tools/osv-linter/internal/pkgchecker/packagist.go

```diff
@@ -0,0 +1,89 @@
+package pkgchecker
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+
+	"github.com/ossf/osv-schema/linter/internal/faulttolerant"
+	"github.com/tidwall/gjson"
+)
+
+var cache sync.Map
```

Comment:
nit: This should just be for packagist, so let's rename this to packagistCache. 

In theory we can put all the ecosystem repo's in here, but I think it's probably clearer to have individual cache vars per ecosystem impl.